### PR TITLE
Add JDK 21 to CI matrix

### PR DESCRIPTION
**Pull request by armanbilge**
Trying to do this separately from https://github.com/typelevel/cats-effect/pull/3870.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -135,11 +136,17 @@ ThisBuild / githubWorkflowPublishPreamble +=
 val OldGuardJava = JavaSpec.temurin("8")
 val LTSJava = JavaSpec.temurin("11")
 val LatestJava = JavaSpec.temurin("17")
+val LoomJava = JavaSpec.oracle("21")
```
Comment (durban) [inline build.sbt:139]: Is there no temurin 21? On adoptium.net there seems to be one...

**Review by durban — APPROVED** [review]

File: build.sbt (hunk 1 above)
Comment (armanbilge) [reply build.sbt:139]: For some reason not yet, I tried that first and changed in https://github.com/typelevel/cats-effect/pull/3870/commits/aeda41c9c4c81a4f2429d240a3433a4eed35c77c.

Comment (vasilmkd) [reply build.sbt:139]: I haven't updated the index yet. It wasn't available as of couple of days ago when I last checked. I'll do it later tonight.

Comment (armanbilge) [reply build.sbt:139]: @vasilmkd actually we are not using the jdk index anymore 😅 

If it's changed recently let me check again.

**Pull request merged**